x-pack/elastic-agent/pkg/agent/application/managed_mode.go

```diff
@@ -205,6 +206,7 @@ func newManaged(
 		[]context.CancelFunc{managedApplication.cancelCtxFn},
 		reexec,
 		acker)
+	uc.SetUpgrader(managedApplication.upgrader)
```

Comment:
should we allow to upgrade agent when enrolled?

Reply:
I think we should allow it, but I can disable it if we think that is correct.